development/tools/pkg/dnscollector/collector.go

```diff
@@ -0,0 +1,237 @@
+			return false, err
+		}
+
+		ipAddressThreshold := time.Since(ipCreationTime).Hours() - float64(minAgeInHours)
+		//TODO: Remove
+		//log.Infof("ipAddressThreshold: %v", ipAddressThreshold)
+		ageMatches = ipAddressThreshold > 0
+
+		return nameMatches && ageMatches, nil
+	}
+}
+
+type addressWrapper struct {
+	data   *compute.Address
+	region string
+}
+
+//New returns an new instance of the Collector
+func New(computeAPI ComputeAPI, dnsAPI DNSAPI, removalPredicate IPAddressRemovalPredicate) *Collector {
+	return &Collector{computeAPI, dnsAPI, removalPredicate}
+}
+
+// Run executes disks garbage collection process
+func (gc *Collector) Run(project string, managedZone string, makeChanges bool) (allSucceeded bool, err error) {
+
+	var msgPrefix string
+	if !makeChanges {
+		msgPrefix = "[DRY RUN] "
+	}
+
+	common.Shout("Looking for matching IP Addresses and DNS Records in project: \"%s\" and zone: \"%s\" ...", project, managedZone)
+
+	matchingIPs, allSucceeded, err := gc.listIPs(project)
+	if err != nil {
+		return false, err
+	}
+
+	if len(matchingIPs) > 0 {
+		log.Infof("%sFound %d matching IP Addresses", msgPrefix, len(matchingIPs))
+		common.Shout("Removing matching IP Addresses / DNS Records ...")
+	} else {
+		log.Infof("%sFound no IP Addresses to delete", msgPrefix)
+	}
+
+	allDNSRecords, err := gc.listDNSRecords(project, managedZone)
+
+	if err != nil {
+		log.Error("listing DNS Records for project \"%s\" and zone \"%s\", %v", project, managedZone, err)
+		return false, err
+	}
+
+	for _, ipAddress := range matchingIPs {
+
+		associatedRecords := findAssociatedRecords(ipAddress.data.Address, allDNSRecords)
+
+		log.Infof("Processing IP Adress: %s, name: \"%s\", %d associated DNS record(s)", ipAddress.data.Address, ipAddress.data.Name, len(associatedRecords))
+
+		for _, dnsRecord := range associatedRecords {
+			if makeChanges {
+				err = gc.dnsAPI.deleteDNSRecords(project, managedZone, dnsRecord)
+				if err != nil {
+					log.Errorf("deleting DNS Records %s: %#v", "DNS", err)
+					allSucceeded = false
+					continue
+				}
+			}
+			log.Infof("%sRequested DNS record delete: \"%s\". Project \"%s\", zone \"%s\"", msgPrefix, dnsRecord.Name, project, managedZone)
+		}
+
+		if makeChanges {
+			err = gc.computeAPI.deleteIPAddress(project, ipAddress.region, ipAddress.data.Name)
+			if err != nil {
+				log.Errorf("deleting IP Address %s: %#v", "IP_ADDRESS", err)
+				allSucceeded = false
+				continue
+			}
+		}
+		log.Infof("%sRequested IP Address delete: \"%s\". Project \"%s\", zone \"%s\", creationTimestamp: \"%s\"", msgPrefix, ipAddress.data.Address, project, managedZone, ipAddress.data.CreationTimestamp)
+	}
+
+	return allSucceeded, nil
+}
+
+//List IP Addresses in all regions. It's a "best effort" implementation,
+func (gc *Collector) listIPs(project string) (res []*addressWrapper, allSucceeded bool, err error) {
+	regions, err := gc.computeAPI.lookupRegions(project, "europe-*")
```

Comment:
Have you considered passing list of zones as argument?

Reply:
I will provide an explicit list of regions as a mandatory input parameter.